devenv/pythons.py

```diff
@@ -1,37 +1,22 @@
 from __future__ import annotations
 
 import os
-import shutil
-import subprocess
+import sys
 
 from devenv.constants import CI
 from devenv.constants import root
 from devenv.lib import archive
 
 
-def _check_version(exe: str, version: str) -> bool:
-    """Returns true if the executable at the given path is a python interpreter
-    with a semver-compatible version."""
+def _is_sys_compatible(version: str) -> bool:
+    """Returns ``True`` if the current python interpreter is semver-compatible
+    with the given version."""
 
-    try:
-        line = (
-            subprocess.check_output([exe, "--version"], text=True)
-            .strip()
-            .lower()
-        )
-    except subprocess.CalledProcessError:
-        return False
-
-    if not line.startswith("python "):
-        return False
-
-    actual = line[7:].split(".")
     expected = version.split(".")
 
     return (
-        len(actual) > 2
-        and actual[:2] == expected[:2]
-        and actual[2] >= expected[2]
+        sys.version_info[:2] == expected[:2]
+        and sys.version_info[2] >= expected[2]
     )
```

Comment:
can just use `sys.version_info >= expected` -- tuple comparisons work out nicely

though you'll have to convert the version to a sequence of integers `expected = [int(part) for part in version.split('.')]`

Reply:
oh yeah, i didn't run that yet. though i want major + minor to match exactly, just patch needs to be more recent. trying to emulate semver semantics

Reply:
ah yeah then I'd do `sys.version_info[:2] == version[:2] and sys.version_info >= version`